src/derivative_operators/convolutions.jl

```diff
@@ -83,45 +95,22 @@ function convolve_BC_left!(x_temp::AbstractVector{T}, _x::BoundaryPaddedVector,
     for i in 1 : A.boundary_point_count
         cur_stencil = stencil[i]
         cur_coeff   = typeof(coeff)   <: AbstractVector ? coeff[i] : true
+        # need to account for x.l in first interior
         xtempi = cur_coeff*cur_stencil[1]*_x.l
         cur_stencil = use_winding(A) && cur_coeff < 0 ? reverse(cur_stencil) : cur_stencil
         @inbounds for idx in 2:A.boundary_stencil_length
             xtempi += cur_coeff * cur_stencil[idx] * _x.u[idx-1]
         end
         x_temp[i] = xtempi
     end
-    # need to account for x.l in first interior
-    mid = div(A.stencil_length,2) + 1
-    x = _x.u
-    i = 1 + A.boundary_point_count
-    xtempi = zero(T)
-    cur_stencil = eltype(A.stencil_coefs) <: AbstractVector ? A.stencil_coefs[i-A.boundary_point_count] : A.stencil_coefs
-    cur_coeff   = typeof(coeff)   <: AbstractVector ? coeff[i-A.boundary_point_count] : true
-    cur_stencil = use_winding(A) && cur_coeff < 0 ? reverse(cur_stencil) : cur_stencil
-    xtempi = cur_coeff*cur_stencil[1]*_x.l
-    @inbounds for idx in 2:A.stencil_length
-        xtempi += cur_coeff * cur_stencil[idx] * x[(i-1) - (mid-idx) + 1]
-    end
-    x_temp[i] = xtempi
```

Comment:
This code is resetting `x_temp[i]` the wrong way. We should use `A.boundary_stencil_length` and not `A.stencil_length`.